src/util/addresses.ts

```diff
@@ -294,10 +294,18 @@ export const WETH9: {
 export const BEACON_CHAIN_DEPOSIT_ADDRESS =
   '0x00000000219ab540356cBB839Cbe05303d7705Fa';
 
-export function getAddress(currency: Currency): string {
+export function getAddressLowerCase(currency: Currency): string {
   if (currency.isToken) {
     return currency.address.toLowerCase();
   } else {
     return ADDRESS_ZERO;
   }
 }
+
+export function getAddress(currency: Currency): string {
```

Comment:
what's the reason to not always use the lowercase?

Reply:
so in case of cached routes, we aren't lower casing the token addresses in routing-api (https://github.com/Uniswap/routing-api/blob/main/lib/handlers/marshalling/token-marshaller.ts#L18). This means the cached routes key in dynamo remains the original form. If we start lower casing the cached routes token addresses, we will have lots of cached routes cache misses for a short period of time.